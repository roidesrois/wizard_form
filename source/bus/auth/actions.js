// Types
import { types } from './types';

export const authActions = {
    // Sync
    initialize: () => ({
        type: types.INITIALIZE,
    }),
    authenticate: () => ({
        type: types.AUTHENTICATE,
    }),
    logout: () => ({
        type: types.LOGOUT,
    }),
    valid: () => ({
        type: types.VALID,
    }),
    registered: () => ({
        type: types.REGISTERED,
    }),

    // Async
    initializeAsync: () => ({
        type: types.INITIALIZE_ASYNC,
    }),
    loginAsync: (credentials) => ({
        type:    types.LOGIN_ASYNC,
        payload: credentials,
    }),
    authenticateAsync: () => ({
        type: types.AUTHENTICATE_ASYNC,
    }),
    signupAsync: (userData) => ({
        type:    types.SIGNUP_ASYNC,
        payload: userData,
    }),
    confirmSignupAsync: (registrationHash) => ({
        type:    types.CONFIRM_SIGNUP_ASYNC,
        payload: registrationHash,
    }),
    activateAccountAsync: (userData) => ({
        type:    types.ACTIVATE_ACCOUNT_ASYNC,
        payload: userData,
    }),
    logoutAsync: () => ({
        type: types.LOGOUT_ASYNC,
    }),
};
