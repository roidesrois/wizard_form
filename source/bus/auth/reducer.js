// Core
import { Map } from 'immutable';

// Instruments
import { types } from './types';

const initialState = Map({
    isInitialized:    false,
    isAuthenticated:  false,
    userIsValid:      false,
    userIsRegistered: false,
});

export const authReducer = (state = initialState, action) => {
    switch (action.type) {
        case types.INITIALIZE:
            return state.set('isInitialized', true);

        case types.AUTHENTICATE:
            return state.set('isAuthenticated', true);

        case types.LOGOUT:
            return state.set('isAuthenticated', false);

        case types.VALID:
            return state.set('userIsValid', true);

        case types.REGISTERED:
            return state.set('userIsRegistered', true);

        default:
            return state;
    }
};
