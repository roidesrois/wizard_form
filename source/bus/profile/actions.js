// Types
import { types } from './types';

export const profileActions = {
    // Sync
    fillProfile: (profile) => ({
        type:    types.FILL_PROFILE,
        payload: profile,
    }),
    updateAvatar: (newAvatarUrl) => ({
        type:    types.UPDATE_AVATAR,
        payload: newAvatarUrl,
    }),
    clearProfile: () => ({
        type: types.CLEAR_PROFILE,
    }),

    // Async
    updateNameAsync: (newName) => ({
        type:    types.UPDATE_NAME_ASYNC,
        payload: newName,
    }),
    updateAvatarAsync: (newAvatar) => ({
        type:    types.UPDATE_AVATAR_ASYNC,
        payload: newAvatar,
    }),
    updatePasswordAsync: (newPassword) => ({
        type:    types.UPDATE_PASSWORD_ASYNC,
        payload: newPassword,
    }),
    updateEmailAsync: (newEmail) => ({
        type:    types.UPDATE_EMAIL_ASYNC,
        payload: newEmail,
    }),
    resetPasswordAsync: (email) => ({
        type:    types.RESET_PASSWORD_ASYNC,
        payload: email,
    }),
};
