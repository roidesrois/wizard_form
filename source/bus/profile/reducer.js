// Core
import { Map } from 'immutable';

// Instruments
import { types } from './types';

const initialState = Map({
    id:        '',
    avatar:    '',
    firstName: '',
    lastName:  '',
    token:     '',
});

export const profileReducer = (state = initialState, action) => {
    switch (action.type) {
        case types.FILL_PROFILE:
            return state.merge(action.payload);

        case types.UPDATE_AVATAR:
            return state.set('avatar', action.payload);

        case types.CLEAR_PROFILE:
            return state.clear();

        default:
            return state;
    }
};
