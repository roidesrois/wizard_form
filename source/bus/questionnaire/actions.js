// Types
import { types } from './types';

export const questionnaireActions = {
    // Sync
    fillQuestionnaire: (questionnaire) => ({
        type:    types.FILL_QUESTIONNAIRE,
        payload: questionnaire,
    }),

    // setQuestionnaireValidation: (validation) => ({
    //     type:    types.SET_QUESTIONNAIRE,
    //     payload: validation,
    // }),

    // Async
    fetchQuestionnaireAsync: () => ({
        type: types.FETCH_QUESTIONNAIRE_ASYNC,
    }),
};
