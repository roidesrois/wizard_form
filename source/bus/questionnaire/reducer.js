// Core
import { fromJS, List } from 'immutable';

// Instruments
import { types } from './types';

const initialState = List();

export const questionnaireReducer = (state = initialState, action) => {
    switch (action.type) {
        case types.FILL_QUESTIONNAIRE:
            return fromJS(action.payload);

        // case types.SET_QUESTIONNAIRE:
        //     return action.payload;

        default:
            return state;
    }
};
