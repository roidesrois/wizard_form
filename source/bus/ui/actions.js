// Core
import { v4 } from 'uuid';

// Instruments
import { types } from './types';

export const uiActions = {
    startFetching: () => ({
        type: types.START_FETCHING,
    }),
    stopFetching: () => ({
        type: types.STOP_FETCHING,
    }),
    emitError: (error, meta = null) => ({
        type:    types.EMIT_ERROR,
        payload: error,
        error:   true,
        meta,
    }),
    showNotification: (message, type = 'info', source = '') => ({
        type:    types.SHOW_NOTIFICATION,
        payload: {
            id: v4(),
            message,
            type,
            source,
        },
    }),
    hideNotification: () => ({
        type: types.HIDE_NOTIFICATION,
    }),
    setOnlineState: () => ({
        type: types.SET_ONLINE_STATE,
    }),
    setOfflineState: () => ({
        type: types.SET_OFFLINE_STATE,
    }),
};
