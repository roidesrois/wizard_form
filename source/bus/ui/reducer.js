// Core
import { Map, fromJS } from 'immutable';

// Instruments
import { types } from './types';

const initialState = Map({
    notification: Map(),
    isOnline:     false,
    isFetching:   false,
});

export const uiReducer = (state = initialState, action) => {
    switch (action.type) {
        case types.START_FETCHING:
            return state.set('isFetching', true);

        case types.STOP_FETCHING:
            return state.set('isFetching', false);

        case types.SHOW_NOTIFICATION:
            return state.set('notification', fromJS(action.payload));

        case types.HIDE_NOTIFICATION:
            return state.update('notification', (notification) => notification.clear());

        case types.SET_ONLINE_STATE:
            return state.set('isOnline', true);

        case types.SET_OFFLINE_STATE:
            return state.set('isOnline', false);

        default:
            return state;
    }
};
