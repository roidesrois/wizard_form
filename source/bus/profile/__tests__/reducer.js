// Core
import { Map } from 'immutable';

// Instruments
import { profileReducer } from '../reducer';
import { profileActions } from '../actions';

const {
    testData: { userProfile, url },
} = global;

const initialState = Map(userProfile);

describe('profile reducer:', () => {
    test('should return initial state by default', () => {
        expect(profileReducer(void 0, {})).toMatchSnapshot();
    });

    test('should handle fillProfile action', () => {
        expect(
            profileReducer(void 0, profileActions.fillProfile(userProfile)),
        ).toMatchSnapshot();
    });

    test('should handle updateAvatar action', () => {
        expect(
            profileReducer(void 0, profileActions.updateAvatar(url)),
        ).toMatchSnapshot();
    });

    test('should handle clearProfile action', () => {
        expect(
            profileReducer(initialState, profileActions.clearProfile()),
        ).toMatchSnapshot();
    });
});
