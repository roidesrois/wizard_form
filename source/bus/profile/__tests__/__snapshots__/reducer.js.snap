// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile reducer: should handle clearProfile action 1`] = `Immutable.Map {}`;

exports[`profile reducer: should handle fillProfile action 1`] = `
Immutable.Map {
  "id": "TEST_ID",
  "avatar": "TEST_AVATAR",
  "firstName": "TEST_FIRST_NAME",
  "lastName": "TEST_LAST_NAME",
  "token": "TEST_TOKEN",
}
`;

exports[`profile reducer: should handle updateAvatar action 1`] = `
Immutable.Map {
  "id": "",
  "avatar": "https://www.url.com",
  "firstName": "",
  "lastName": "",
  "token": "",
}
`;

exports[`profile reducer: should return initial state by default 1`] = `
Immutable.Map {
  "id": "",
  "avatar": "",
  "firstName": "",
  "lastName": "",
  "token": "",
}
`;
